redash/utils/__init__.py

```diff
@@ -42,6 +42,10 @@ def dt_from_timestamp(timestamp, tz_aware=True):
 def slugify(s):
     return re.sub('[^a-z0-9_\-]+', '-', s.lower())
 
+def gen_signature_hash(parameters, key):
+    if key is None:
+        key = settings.EMBED_KEY
```

Comment:
I think it will be safer if we use a distinct key for each query. We can reuse the current query API key, we just need to make sure we don't send it to "guest" users (it might be returned by default in the query API response now).